lib/puppet/provider/package/windows.rb

```diff
@@ -64,7 +64,7 @@ def install
 
     command = [installer.install_command(resource), install_options].flatten.compact.join(' ')
     working_dir = File.dirname(resource[:source])
-    if !Puppet::FileSystem.exist?(working_dir) && resource[:source] =~ /\.msi"?\Z/i
+    if !Puppet::FileSystem.exist?(working_dir) && (resource[:source] =~ /\.msi"?\Z/i || resource[:source] =~ /\.exe"?\Z/i)
```

Comment:
I'm not sure this check `(resource[:source] =~ /\.msi"?\Z/i || resource[:source] =~ /\.exe"?\Z/i)` is still needed, I think it will always be true.